simtools/model/array_model.py

```diff
@@ -1,19 +1,473 @@
+
+        arrayConfigData = collectDataFromYamlOrDict(arrayConfigFile, arrayConfigData)
+        self._loadArrayData(arrayConfigData)
+
+        self._setConfigFileDirectory()
+
+        self._buildArrayModel()
+        # End of init
+
+    @property
+    def numberOfTelescopes(self):
+        return self.layout.getNumberOfTelescopes()
+
+    def _loadArrayData(self, arrayConfigData):
+        ''' Load parameters from arrayData.
+
+        Parameters
+        ----------
+        arrayConfigData: dict
+        '''
+        # Validating arrayConfigData
+        # Keys 'site', 'layoutName' and 'default' are mandatory.
+        # 'default' must have 'LST', 'MST' and 'SST' (for South site) keys.
+        self._validateArrayData(arrayConfigData)
+
+        # Site
+        self.site = names.validateSiteName(arrayConfigData['site'])
+
+        # Grabing layout name and building LayoutArray
+        self.layoutName = names.validateLayoutArrayName(arrayConfigData['layoutName'])
+        self.layout = LayoutArray.fromLayoutArrayName(
+            self.site + '-' + self.layoutName,
+            label=self.label
+        )
+
+        # Model version
+        if 'modelVersion' not in arrayConfigData.keys() or arrayConfigData['modelVersion'] is None:
+            self._logger.warning('modelVersion not given in arrayConfigData - using current')
+            self.modelVersion = 'current'
+        else:
+            self.modelVersion = names.validateModelVersionName(arrayConfigData['modelVersion'])
+
+        # Removing keys that were stored in attributes and keepig the remaining as a dict
+        self._arrayConfigData = {
+            k: v for (k, v) in arrayConfigData.items()
+            if k not in ['site', 'layoutName', 'modelVersion']
+        }
+    # End of _loadArrayData
+
+    def _validateArrayData(self, arrayConfigData):
+        '''
+        Validate arrayData by checking the existence of the relevant keys.
+        Searcing for the keys: 'site', 'array' and 'default'
+        And the sub-keys in 'default': 'LST', 'MST' and 'SST', in case of
+        South array.
+        '''
+
+        def runOverPars(pars, data, parent=None):
+            '''Run over pars and validate it. '''
+            allKeys = data.keys() if parent is None else data[parent].keys()
+            for pp in pars:
+                if pp not in allKeys:
+                    key = pp if parent is None else parent + '.' + pp
+                    msg = (
+                        'Key {} was not found in arrayConfigData '.format(key)
+                        + '- impossible to build array model'
+                    )
+                    self._logger.error(msg)
+                    raise InvalidArrayConfigData(msg)
+
+        runOverPars(['site', 'layoutName', 'default'], arrayConfigData)
+        runOverPars(['LST', 'MST'], arrayConfigData, parent='default')
+        if names.validateSiteName(arrayConfigData['site']) == 'South':
```

Comment:
I don't think we should hard wire the telescope types for North and South into the code. Please check if there isn't another good way to do this.

Reply:
As I replied on another comment, we could actually check whether all the telescopes listed in the layout have a tel model given individually, and if not, we search for the default and raise an error when it is not there.

In this way, the default tel models won't be required anymore and we don't need to know which tel sizes apply for each site.

I will implement that later. It seems like a better option.

Reply:
Done in https://github.com/gammasim/gammasim-tools/pull/110/commits/1bc0416bc8ec6114b8a95222dd14badbdf9a8ddc